ome_zarr/writer.py

```diff
@@ -306,6 +306,18 @@ def write_multiscales_metadata(
             if axes is not None:
                 ndim = len(axes)
 
+    if isinstance(metadata, dict) and "omero" in metadata["metadata"]:
+        assert isinstance(metadata["metadata"], dict), "metadata must be a `dict`"
```

Comment:
This `assert` will never fail since you check that it's true on the previous line